### DISCUSSION TITLE
Bucket events when depositing an archive & snowball decompression

### DISCUSSION THREAD

**Discussion by comassky** [minio/minio#17759] [Q&A]
Hello,

With a view to implementing the "Change Data Capture" pattern, I'm using MinIO's event functionality by issuing PUT / GET / DELETE events in a Kafka topic.

This works very well for unitary files.

I also use tar & tar.gz file upload, with decompression via SNOWBALL and the snowball-auto-extract and minio-snowball-prefix options.

In this case, the files are decompressed with a prefix, but the PUT events are not pushed to Kafka.

Is it possible to trigger the sending of events for each file decompressed from the deposited archive?

Regards

**Comment by klauspost** [comment]
Let me see if that is a feasible change.

**Comment by comassky** [comment]
Thank you!
Just to clarify, uploading the tar doesn't cause an event either.

To sum up, an upload of a tar file with snowball decompression does not generate any event (neither for the tar dump, nor for the decompressed files).

**Reply by klauspost** [reply]
Uploading the TAR will generate an audit log, as expected, but individual files do not generate events.

Sent PR: https://github.com/minio/minio/pull/17760

**Reply by comassky** [reply]
Ok so it's not possible to trigger audit log for each file ?

If my tar contains 2 files, i hope to have 2 audit logs.

Maybe an option or an header to add  ?

**Reply by klauspost** [reply]
Audit logs are on a per request basis, and not per "operation".

**Reply by comassky** [reply]
Okay, and do we have in the tar audit log all the sub-files present ?

**Reply by klauspost** [reply]
An audit entry [can contain multiple objects](https://github.com/minio/pkg/blob/main/logger/message/audit/entry.go#L45).

Without an individual size, I don't suppose it is of much use. Either way, adding it will be a much bigger task.
